TrackletAlgorithm/MatchEngineUnit.h

```diff
@@ -205,138 +315,60 @@ inline int getIPhi() {
-	 stubmask_[index]=0;
-	 second_ =  index[0];
-	 phiPlus_ =  index[1];
-	 nstubs_ = nstubsall_[index];
-       }
-     } else {
-       istub_++;
-     }
-   }
-
-   //Read stub memory and extract data fields
-   ap_uint<10> stubadd=(iphiSave,zbin,istubtmp);
-   stubdata_ = stubmem[bx&1][stubadd];
-   projfinephi__ = projfinephi_;
-   projfinezadj_ = projfinezadj;
-   isPSseed_ = isPSseed;
-   projrinv_ = projrinv;
-   projbuffer__ = projbuffer_;
-   good_ =  process;
+inline MATCH peek() {
+#pragma HLS inline  
+  return matches_[readindex_];
+}
+ 
+inline void advance() {
+#pragma HLS inline  
+  readindex_++;  
+}
 
-   
- } // end step
+ private:
 
- bool Good_() const { return good_;}
- bool Good__() const { return good__;}
+ //Buffers for the matches
+ INDEX writeindex_, readindex_;
+ MATCH matches_[1<<MatchEngineUnitBase<VMProjType>::kNBitsBuffer];
+ ap_uint<kNBits_MemAddr> projseqs_[1<<MatchEngineUnitBase<VMProjType>::kNBitsBuffer];
 
- private:
- INDEX writeindex_;
- INDEX readindex_;
+ //Current projection
  ap_uint<4> nstubsall_[4];
- NSTUBS nstubs_;
+ NSTUBS nstubs_, istub_;
  ap_uint<4> stubmask_;
  ap_uint<1> second_;
  ap_uint<1> phiPlus_;
  ap_int<2> shift_;
  bool idle_;
- int ivmphi;
- int unit_;  // only used for debugging to identify MEU
  ap_uint<3> iphi_;
- BXType bx;
+ BXType bx_;
  bool empty_;
- NSTUBS istub_=0;
- VMStubMECM<VMSMEType> stubdata_, stubdata__; 
- bool good_, good__;
- ap_int<5> projfinephi__, projfinephi___;
- ap_int<5> projfinezadj_, projfinezadj__;
- bool isPSseed_, isPSseed__;
- VMProjection<BARREL>::VMPRINV projrinv_, projrinv__;
- ProjectionRouterBuffer<BARREL, AllProjectionType> projbuffer__, projbuffer___;
+ VMStubMECM<VMSMEType> stubdata__, stubdata___; 
+ typename ProjectionRouterBuffer<BARREL, AllProjectionType>::TCID tcid_;
+ ProjectionRouterBuffer<BARREL, AllProjectionType> projbuffer_;
+ ap_uint<kNBits_MemAddr> projseq_;
+ bool isPSseed_;
+
 
- MATCH matches_[1<<MatchEngineUnitBase<VMProjType>::kNBitsBuffer];
- ap_int<5> projfinezadj; //FIXME Need replace 5 with const
- ap_int<5> projfinephi_; //FIXME Need replace 5 with const
- typename ProjectionRouterBuffer<BARREL, AllProjectionType>::TCID tcid;
- bool isPSseed;
  typename ProjectionRouterBuffer<BARREL, AllProjectionType>::VMPZBINNOFLAG zbin;
- VMProjection<BARREL>::VMPRINV projrinv;
- VMProjection<BARREL>::VMPID projindex;
- ProjectionRouterBuffer<BARREL, AllProjectionType> projbuffer_;
- ap_uint<(1 << (2 * MatchEngineUnitBase<VMProjType>::kNBitsBuffer))> nearFullLUT = nearFull3Unit<MatchEngineUnitBase<VMProjType>::kNBitsBuffer>();
- bool isSecond[4] = {false, true, false, true};
- bool isPhiPlus[4] = {false, false, true, true};
- ap_uint<1<<(kNBits_MemAddrBinned+2)> onlyOneStub = hasOneStub<kNBits_MemAddrBinned+2>();
+
+ //Pipeline variables
+ bool good__, good___;
+ ap_int<5> projfinephi__, projfinephi___; //FIXME Need replace 5 with const
+ ap_int<5> projfinezadj__, projfinezadj___; //FIXME Need replace 5 with const
```

Comment:
@aryd Can you replace 5 with a meaningfully named constant or the type with a `typedef`?